test-framework/core/src/main/java/org/keycloak/testframework/admin/AdminClientFactorySupplier.java

```diff
@@ -1,17 +1,27 @@
 package org.keycloak.testframework.admin;
 
 import org.keycloak.testframework.annotations.InjectAdminClientFactory;
+import org.keycloak.testframework.https.ManagedCertificates;
 import org.keycloak.testframework.injection.InstanceContext;
 import org.keycloak.testframework.injection.RequestedInstance;
 import org.keycloak.testframework.injection.Supplier;
 import org.keycloak.testframework.server.KeycloakServer;
 
+import java.security.KeyStore;
+
 public class AdminClientFactorySupplier implements Supplier<AdminClientFactory, InjectAdminClientFactory> {
 
     @Override
     public AdminClientFactory getValue(InstanceContext<AdminClientFactory, InjectAdminClientFactory> instanceContext) {
         KeycloakServer server = instanceContext.getDependency(KeycloakServer.class);
-        return new AdminClientFactory(server.getBaseUrl());
+
+        if (!server.isTlsEnabled()) {
+            return new AdminClientFactory(server.getBaseUrl());
+        }
+
+        ManagedCertificates managedCert = instanceContext.getDependency(ManagedCertificates.class);
+        KeyStore serverKeyStore = managedCert.getKeyStore();
+        return new AdminClientFactory(server.getBaseUrl(), serverKeyStore);
```

Comment:
Nitpick, but I find it cleaner to read with this:
```suggestion
        } else {
            ManagedCertificates managedCert = instanceContext.getDependency(ManagedCertificates.class);
            KeyStore serverKeyStore = managedCert.getKeyStore();
            return new AdminClientFactory(server.getBaseUrl(), serverKeyStore);
       }
```